crates/cli/src/main.rs

```diff
@@ -209,27 +208,42 @@ fn init_db(command: &ContenderSubcommand) -> Result<(), CliError> {
 }
 
 fn init_tracing() {
-    let filter = EnvFilter::try_from_default_env().ok(); // fallback if RUST_LOG is unset
-    #[cfg(feature = "async-tracing")]
-    {
-        use tracing_subscriber::{fmt, layer::SubscriberExt, util::SubscriberInitExt, Layer};
-        let tokio_layer = console_subscriber::ConsoleLayer::builder()
-            .with_default_env()
-            .spawn();
-        let fmt_layer = fmt::layer()
-            .with_ansi(true)
-            .with_target(true)
-            .with_line_number(true)
-            .with_filter(filter);
-
-        tracing_subscriber::Registry::default()
-            .with(fmt_layer)
-            .with(tokio_layer)
-            .init();
+    use tracing_subscriber::{
+        fmt, layer::SubscriberExt, util::SubscriberInitExt, EnvFilter, Layer,
+    };
+
+    let rust_log = std::env::var("RUST_LOG").unwrap_or_default().to_lowercase();
+    let debug_mode = rust_log.contains("=debug");
+
+    let filter = EnvFilter::try_from_default_env().unwrap_or_else(|_| EnvFilter::new("info"));
+
+    let fmt_layer = fmt::layer()
+        .with_ansi(true)
+        .with_target(debug_mode)
+        .with_line_number(debug_mode)
+        .with_file(debug_mode)
+        .with_filter(filter);
+
+    tracing_subscriber::registry().with(fmt_layer).init();
+}
+
+#[cfg(test)]
+mod tests {
+    use temp_env::with_var;
+
+    #[test]
+    fn debug_mode_detected_from_rust_log() {
+        with_var("RUST_LOG", Some("contender=debug"), || {
```

Comment:
I don't think this test is really necessary, all it does is test the `with_var` function.